zettarepl/zettarepl.py

```diff
@@ -413,3 +435,8 @@ def _run_remote_retention(self, now: datetime):
                 logger.warning("Remote retention failed on %r: error destroying snapshots: %r",
                                transport, e)
                 continue
+
+    def _get_removal_dates(self):
+        from middlewared.client import Client
+        with Client() as c:
```

Comment:
I wonder if we should isolate interaction with middleware in another module perhaps ?